### PR TITLE
Update henson logging to allow propagate to be configurable

### PR DESCRIPTION
**Pull request by boguth**
https://ihm-it.atlassian.net/browse/IHRACP-3480

tl;dr - we're duplicating logs because we have a parent logger writing to stderr and a child logger writing to stdout. Turning propagate to false keeps events from being passed from the child logger to the parent logger so that we only output logs once to stdout. 

Service applications implement the following code: 
```py
    app = Application(name) 
```
which calls this: 
```py
# instantiates a base python logger which inherits from the root logger.
# also implicitly sets a StreamHandler to the root logger which outputs to stderr
self.logger = logging.getLogger(self.name) # henson/base.py:64
```

Next, we typically call this in our service applications: 
```py
logger.init_app(app)
```
which causes any subsequent invocations of `logger` to invoke code which on first pass does this: 
```py
    # henson_logging/__init__.py:130
    'handlers': {
        'henson': {
            **self.app.settings['LOG_HANDLER_KWARGS'], <-- sets output to stdout
            'class': self.app.settings['LOG_HANDLER'],
            'formatter': self.app.settings['LOG_FORMATTER'],
        },
    }, 
    
   ....

  # henson_logging/__init__.py:147
  structlog.configure(
      processors=self.app.settings['LOG_PROCESSORS'],
      context_class=self.app.settings['LOG_CONTEXT_CLASS'],
      logger_factory=self.app.settings['LOG_FACTORY'], # 
     #^^^^^^ binds any subsequent structlog.get_logger calls to stdlib, which inherits from the root logger https://www.structlog.org/en/stable/api.html#structlog.stdlib.LoggerFactory or https://www.structlog.org/en/stable/getting-started.html#structlog-and-standard-library-s-logging
      wrapper_class=self.app.settings['LOG_WRAPPER_CLASS'],
      cache_logger_on_first_use=True,
  )

  self._logger = structlog.get_logger(self.app.name).bind()
```

This result of all this means that we now have two StreamHandlers, one handling events emitted by our logger and one handling events emitted by our base python logger - both of which inherit from the root logger. 

Each logger has a propagate setting which defaults to true. If it is true then:
```

 ...events logged to this logger will be passed to the handlers of higher level (ancestor) loggers, in addition to any handlers attached to this logger. Messages are passed directly to the ancestor loggers’ handlers - neither the level nor filters of the ancestor loggers in question are considered.

```
Reference: https://docs.python.org/3/library/logging.html#logging.Logger.propagate. 

This was causing us to outputs logs twice. Once to stdout and then to stderr. If we set propogate to false, this issue will stop. So I made that a configurable option.

### CODE REVIEW

**Review by jnguyengit — CHANGES_REQUESTED** [review]
Need to update change logs and versioning if we want this to go out right away. 
`setup.py` -> https://github.com/iheartradio/Henson-Logging/blob/master/setup.py#L23
`changes.rst` -> https://github.com/iheartradio/Henson-Logging/blob/master/CHANGES.rst

File: setup.py [hunk 1]
```diff
@@ -20,7 +20,7 @@ def read(filename):
 
 setup(
     name='Henson-Logging',
-    version='0.5.0',
+    version='0.6.0',
     author='Andy Dirnberger, Jon Banafato, Leonard Bedner, and others',
```
Comment (Christian-Paul) [inline setup.py:24]: Add your name here and in AUTHORS.rst

**Review by Christian-Paul — CHANGES_REQUESTED** [review]

File: setup.py (hunk 1 above)
Comment (boguth) [reply setup.py]: Done

File: henson_logging/__init__.py [hunk 2]
```diff
@@ -58,6 +58,7 @@ class Logging(Extension):
         'LOG_CONTEXT_CLASS': dict,
         'LOG_FACTORY': structlog.stdlib.LoggerFactory(),
         'LOG_WRAPPER_CLASS': structlog.stdlib.BoundLogger,
+        'PROPAGATE': True,
```
Comment (adityaghosh) [inline henson_logging/__init__.py:61]: Just a comment: Does it make sense to call this `LOG_PROPAGATE` to be in line with all other settings 🤔 .

Comment (boguth) [reply henson_logging/__init__.py:61]: Yeah I like that

**Review by adityaghosh — APPROVED** [review]

File: henson_logging/__init__.py [hunk 3]
```diff
@@ -74,8 +75,18 @@ def init_app(self, app):
                 initialize.
         """
         super().init_app(app)
+        # lines 83-91 use the state of Henson base
```
Comment (e-agbanyo) [inline henson_logging/__init__.py:78]: Remove the line number mentions, it would be confusing if those line shift from code changes in the future. Just leaving a reference to what the code is doing, not where it is, works.

**Review by e-agbanyo — CHANGES_REQUESTED** [review]
Remove the line number mentions, if these change in the future

File: henson_logging/__init__.py (hunk 3 above)
Comment (boguth) [reply henson_logging/__init__.py]: done

**Review by jnguyengit — APPROVED** [review]

**Review by adityaghosh — APPROVED** [review]

**Review by Christian-Paul — APPROVED** [review]

File: setup.py [hunk 4]
```diff
@@ -20,8 +20,8 @@ def read(filename):
 
 setup(
     name='Henson-Logging',
-    version='0.5.0',
-    author='Andy Dirnberger, Jon Banafato, Leonard Bedner, and others',
+    version='0.6.0',
+    author='Andy Dirnberger, Jon Banafato, Leonard Bedner, Bo Guthrie, and others',
```
Comment (lbedner) [inline setup.py:24]: @boguth you're big time now, baby!

**Review by lbedner — APPROVED** [review]

File: setup.py (hunk 4 above)
Comment (boguth) [reply setup.py:24]: Heck yeah I am.

**Review by e-agbanyo — APPROVED** [review]
